src/index.ts

```diff
@@ -44,3 +45,62 @@ export function getServiceData(apiName: string, beaconName: string, chain: strin
     },
   };
 }
+
+interface PrivateKeySender {
+  privateKey: string;
+}
+
+interface MnemonicSender {
+  mnemonic: string;
+  derivationPath?: string;
+}
+
+type SenderAccount = PrivateKeySender | MnemonicSender;
+
+export function isPrivateKeySender(senderAccount: SenderAccount): senderAccount is PrivateKeySender {
+  return Object.prototype.hasOwnProperty.call(senderAccount, 'privateKey');
+}
+
+interface WhitelistBeaconReaderResult {
+  beaconId: string;
+  beaconReaderAddress: string;
+  expirationTimestamp: number;
+  indefiniteWhitelistStatus: boolean;
+}
+
+export async function whitelistBeaconReader(
+  beaconId: string,
+  beaconReaderAddress: string,
+  chain: string,
+  providerUrl: string,
+  senderAccount: SenderAccount
+): Promise<WhitelistBeaconReaderResult> {
+  const goBeaconWhitelister = goSync(() =>
+    require(`@api3dao/utility-contracts/SelfServeRrpBeaconServerWhitelister/deployments/${chain}/SelfServeRrpBeaconServerWhitelister.json`)
+  );
+  if (!isGoSuccess(goBeaconWhitelister)) throw new Error(`No whitelister contract found for chain: ${chain}`);
+  const beaconWhitelister = goBeaconWhitelister.data;
+
+  const provider = new ethers.providers.JsonRpcProvider(providerUrl);
+  const wallet = isPrivateKeySender(senderAccount)
+    ? new ethers.Wallet(senderAccount.privateKey)
+    : ethers.Wallet.fromMnemonic(senderAccount.mnemonic, senderAccount.derivationPath);
+  const connectedWallet = wallet.connect(provider);
+  const whitelisterContract = new ethers.Contract(beaconWhitelister.address, beaconWhitelister.abi).connect(
+    connectedWallet
+  );
+
+  await whitelisterContract.whitelistReader(beaconId, beaconReaderAddress);
+  const filter = whitelisterContract.filters.WhitelistedReader(beaconId, beaconReaderAddress);
+  // TODO: Simplify
+  const { args } = await new Promise((resolve) => provider.once(filter, resolve)).then((rawLog) =>
+    whitelisterContract.interface.parseLog(rawLog as ethers.Event)
+  );
+
+  return {
```

Comment:
I do not want to return response from ethers (especially because of BigIntegers it returns). 
I wonder If I should rename the object keys (currently they follow the event names from chain).

Reply:
I think it's fine the way it is and the script calling this function can prettify the output to the end user